docs/rules/no-constructor-params.md

```diff
@@ -0,0 +1,30 @@
+# Disallows constructor parameters in custom elements (no-constructor-params)
+
+Ensures that custom element constructors have no parameters as they
+are never constructed directly but are rather constructed automatically when
+appended to the DOM (or via `createElement` etc).
+
+## Rule Details
+
+This rule disallows constructor parameters in custom element classes.
+
+The following patterns are considered warnings:
+
+```ts
+class Foo extends HTMLElement {
+  constructor(p1, p2) {}
+}
+```
+
+The following patterns are not warnings:
+
+```ts
+class Foo extends HTMLElement {
+  constructor() {}
+}
+```
+
+## When Not To Use It
+
+If you wish to allow parameters to custom element constructors, you should not
```

Comment:
These when not to use sections seem mostly tautological. Should there be some opinion here, like "you should usually keep this rule enabled, as constructor parameters are never passed from the browser. You should only disable it if you need to provide arguments from direct constructor calls, like `new MyElement(a, b)`, and then yo should ensure that the element can handle cases when the arguments aren't provided."

This brings to mind a possibly useful variant of this rule which is that if you _do_ have ctor params, they should be optional.

Reply:
yeah these sections are rarely used well, leftover from trying to follow the same structure as eslint's own rules. ill update it for now and open an issue for supporting optional params